8336753: Don't run serviceability/sa/ClhsdbDumpheap.java with -Xcomp

dlunde opened the pull request:
### Issue Summary

We are seeing quite a few timeouts for `serviceability/sa/ClhsdbDumpheap.java` running with -Xcomp in testing, possibly related to [JDK-8324241](https://bugs.openjdk.org/browse/JDK-8324241). We should disable running `ClhsdbDumpheap.java` with -Xcomp, at least for now.

### Changeset

- Add `@requires vm.compMode != "Xcomp"` to `ClhsdbDumpheap.java`.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8336753](https://bugs.openjdk.org/browse/JDK-8336753): Don't run serviceability/sa/ClhsdbDumpheap.java with -Xcomp (**Sub-task** - P4)


### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) 🔄 Re-review required (review applies to [05eddeb1](https://git.openjdk.org/jdk/pull/20237/files/05eddeb1352e24a82d458fc9b9390e6effa44ff6))
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) 🔄 Re-review required (review applies to [05eddeb1](https://git.openjdk.org/jdk/pull/20237/files/05eddeb1352e24a82d458fc9b9390e6effa44ff6))
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20237/head:pull/20237` \
`$ git checkout pull/20237`

Update a local copy of the PR: \
`$ git checkout pull/20237` \
`$ git pull https://git.openjdk.org/jdk.git pull/20237/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20237`

View PR using the GUI difftool: \
`$ git pr show -t 20237`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20237.diff">https://git.openjdk.org/jdk/pull/20237.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20237#issuecomment-2236813199)

dlunde commented:
/label remove serviceability
/label add hotspot-compiler

vnkozlov reviewed (approved):
Good.

plummercj commented:
/label add serviceability

plummercj commented:
I think the proper way to do this would be to add the test to ProblemList-Xcomp.txt, not by using `@requires`, unless we have determined that the nature of the test suggests that it is not appropriate to run with -Xcomp (I haven't seen any evidence of this. yet). That fact that the issue is new would suggest that there is some underlying bug that needs to be better understood.

However, I'm not so sure even problem listing is the right thing to do at this point. I don't see any cases of this test timing out in our CI and I don't see any links in the CR to runs that are failing. I do see mach5 adhoc runs failing with the timeout, but they all contains user diffs that could also be related to the cause.

vnkozlov commented:
We don't run this test in CI with the same configuration as in our pre-integration testing.

First, we run with "-Xcomp -ea -esa -XX:CompileThreshold=100" flags in CI instead of simple "-Xcomp".
Second, we used to be running this test in tier4 on Linux and Tier6 on windows in CI. But now we are running only in tier6 with `any-x64` hosts for which infrastructure always selects linux-x64.  On other hand we see timeouts on slow machines macOS-x64 and some Windows machines.

vnkozlov commented:
This change similar to what we did for other timeout tests recently.

vnkozlov commented:
And finally I don't think we should run this test with -Xcomp - there are no any benefits from it. This test dumps java heaps.

plummercj commented:
Does not running with `-Xcomp` mean not running it with graal?

vnkozlov commented:
> Does not running with `-Xcomp` mean not running it with graal?

`-Xcomp` is not related to graal at all. It triggers all JIT compilers compilation for each called Java method. And Java thread waits until compilation is finished. It is very slow application execution. But it stress JIT compilers since they do a lot more compilations.

plummercj commented:
Given that there is no inherent incompatibility with `-Xcomp` and the test is capable of running with `-Xcomp`, I suggest that you add an `@comment` before the `@require` stating why it shouldn't be run with `-Xcomp`.

vnkozlov commented:
> Given that there is no inherent incompatibility with `-Xcomp` and the test is capable of running with `-Xcomp`, I suggest that you add an `@comment` before the `@require` stating why it shouldn't be run with `-Xcomp`.

Agree.

TobiHartmann commented:
FTR, I converted this to a sub-task of [JDK-8336805](https://bugs.openjdk.org/browse/JDK-8336805) which keeps track of the regression from [JDK-8324241](https://bugs.openjdk.org/browse/JDK-8324241) that we think is responsible for this.

dlunde commented:
Thanks for the input everyone. I've now added a comment based on the discussion above, and I also added this issue number to the test for tracking purposes.

TobiHartmann commented on test/hotspot/jtreg/serviceability/sa/ClhsdbDumpheap.java:
```diff
@@ -37,9 +37,13 @@
 /**
  * @test
- * @bug 8240989
+ * @bug 8240989 8336753
```
I don't think that change makes sense because the test is not a regression test for JDK-8336753.

TobiHartmann replied:
I don't think that's necessary, one can always find the changeset that added a specific line/comment via the git history.

dlunde replied:
Fair enough, now removed.

TobiHartmann reviewed (approved):
Looks good to me otherwise.

vnkozlov reviewed (approved):
Last update is good.

plummercj reviewed (commented):
Copyright needs updating.

dlunde commented:
> Copyright needs updating.

Thanks, done.

dlunde commented:
Thanks for the reviews. Please sponsor!

/integrate

vnkozlov commented:
/sponsor
